src/classes/OPP_StageToStateMapping_API.cls

```diff
@@ -0,0 +1,148 @@
+/*
+    Copyright (c) 2019, Salesforce.org
+    All rights reserved.
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Opportunity
+* @group-content ../../ApexDocContent/Opportunity.htm
+* @description API Class to allow caller to determine Opportunity's State (Uncommitted, Committed. or Finalized) using the Opportunities Stage and Household Settings
+*/
+public class OPP_StageToStateMapping_API {
```

Comment:
I think this class would be better implemented as an instance class rather than all static classes, so something along the lines of:
```
    OPP_StageToStateMapping stageMapper = new OPP_StageToStateMapping(String stageName);
    String oppState = stageMapper.getOppState();
```

That said, depending on how this will be used in Abacus, maybe the class method is better as:
`String oppState = stageMapper.getOppStateFromStageName(String stageName);`

Reply:
Just curious.  Why do you believe this to be better?  We would be calling it over and over for different opportunities, so it would be extra overhead to repeatedly instantiate to get there...

Reply:
I suppose this depends on the use case. If you're making this call in a loop, I'd go with the model that instantiates the class once outside the loop (i.e., no parameters in the constructor), and then have a class method for `getOppStateFromStageName` that accepts a stage name string. Instantiation that doesn't do much in the constructor isn't terribly expensive either, unless of course it's happening 10000 times in a loop.

On the instance vs. static class, I think the instance class will offer more flexibility down the road if there is other related logic. Static classes should generally be for utility methods. To me this isn't necessarily a _utility_ method. 

The above noted, I could see where the class constructor accepts an Opportunity object and has a method to return the OppStage. That could eventually allow for new methods that return other needed details about the Opp that require some related translation (all TBD based on usage).